File: springrest/src/main/java/it/sistemiLab/dto/UserDTO.java
package it.sistemiLab.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import it.sistemiLab.model.Progetto;
import it.sistemiLab.model.User.Usertype;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO della classe User. Ha gli stessi attributi di User
 * 
 * 
 * 
 *@see User
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class UserDTO {

	private long id;
	
	private String username;
	
	private String password;

	private  String nome;

	private String cognome;

	private Usertype usertype;


}

File: angular/src/dto/userdto.ts
import {Usertype} from './usertype';
import {ClienteDTO} from './clientedto';


export class UserDTO {

   id: number;

   username: string;

   password: string;

   nome: string;

   cognome: string;
   
   cliente: ClienteDTO[];

   usertype: Usertype;



}

